### ISSUE TITLE
Invalid SnapRequest sent by mobile sdk

### ISSUE THREAD

**Issue by imrenagi** [veritrans/go-midtrans#39]
Hi, 

I'm using this module with version (v0.0.0-20190513104831-ff7f567d2249) in my golang backend. However, when we receive the snap request from our android app, we found out that `credit_card.authentication` sent by the android sdk is not a boolean, but string. Meanwhile in go-midtrans module, it is defined as `boolean` in [here](https://github.com/veritrans/go-midtrans/blob/9d3aefaf0e9ef61c60f95c861736217ad61b0a33/request.go#L63).

We are using struct `SnapReq` to decode the json sent by the android and to do some validation against the request sent by android app. However we got error due to the issue i explained above.
```
decoder := json.NewDecoder(r.Body)
var snapRequest go-midtrans.SnapReq
err := decoder.Decode(&snapRequest)
if err != nil {
.....
}
```

Is this expected?

**Comment by rizdaprasetya** [comment]
Actually the `credit_card.authentication` API wise both accept string & boolean, to maximize compatibility. The quick solution will be to change the type to `string`, which should be safe for both:
https://github.com/veritrans/go-midtrans/blob/9d3aefaf0e9ef61c60f95c861736217ad61b0a33/request.go#L63

Or any other alternative to define type that both accept `bool` and `string` from JSON on Go?

**Comment by imrenagi** [comment]
I dont think so.. is it possible to promote this changes on the new version of this library?

**Comment by rizdaprasetya** [comment]
Yes make sense, this should be changed to `string` for next version. But first will need to make proper versioning of this library. Currently we're prioritizing some other development, we'll get back to this once we have proper allocation. 
We'll leave this issue open for now, Pull Request from anyone is welcome to address this.

**Comment by imrenagi** [comment]
Yep. Adding version number will be a good thing to have. I will try to make a PR for changing this to string tho. :D
